src/swap-helpers.ts

```diff
@@ -32,10 +34,21 @@ export function ensureInFuture(
   return target < date.valueOf() ? date : new Date(target)
 }
 
-export interface SwapOrder {
+interface SwapOrderSpendInfo {
+  spendInfo: EdgeSpendInfo
+}
+
+interface SwapOrderMakeTx {
+  makeTxParams: MakeTxParams
+}
+
+type SwapOrderInner = SwapOrderMakeTx | SwapOrderSpendInfo
+
+export type SwapOrder = SwapOrderInner & {
```

Comment:
I'll remove the throw, but it would be good to keep the type structure to ensure only one type is passed. I originally had a second param in SwapOrderMakeTx but removed it before the PR.